### PR TITLE
Updated NEWS file for 1.17

### PR DESCRIPTION
**Pull request by shasson5**
## What
Updated NEWS file for 1.17

## Why ?
v1.17 Release

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -11,9 +11,98 @@
...
+#### Shared Memory
+* Fixed occasional crash when shm_unlink fails during interface initialization
+#### UCS
+* Fixed system device distance calculation for devices with different root
+* Fixed support for large size arrays in ucs_array
+* Fixed synchronization issue in rcache
+#### Tests
+* Fixed test failures when GPU is present but disabled
+* Fixed Active Message hanging issue in ucp_client_server
+* Fixed potential crash due to redundent munmap call in ucp mmap tests
```
Comment (gleon99) [inline NEWS:99]: redundant

File: NEWS [hunk 2]
```diff
@@ -11,9 +11,98 @@
...
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for protocol variants
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols 
+#### RDMA CORE (IB, ROCE, etc.)
+* Added UCX_REVERSE_SL env var to allow configuring reverse SL and set default to UCX_IB_SL value
+* Added support for GID auto-detection in Floating LID based routing
```
[the remark this answers is not in the document]
Comment (shasson5) [reply NEWS:30]: It's a term (FLID)

File: NEWS [hunk 3]
```diff
@@ -11,9 +11,98 @@
 ### Features:
 ### Bugfixes:
 
-## 1.17.0-rc1 (April 19)
-TBD
-
+## 1.17.0-rc1 (May 9, 2024)
```
Comment (yosefe) [inline NEWS:14]: IMO it should be RC2 now

File: NEWS [hunk 4]
```diff
@@ -11,9 +11,98 @@
...
+* Added support for GID auto-detection in Floating LID based routing
+* Imporved performance of RC transport when hardware tag matching is not available
+* Added support for multithreading KSM registration of unaligned buffers
+* Added IB_SEND_OVERHEAD and MM_[SEND|RECV]_OVERHEAD configuration variables
+#### GPU (CUDA, ROCM)
+* Added support for oneAPI Level-Zero library for intel GPUs
+#### UCS
+* Added support for rcache dynamic region alignment
+* Added dynamic bitmap data structure
+* Added support for advanced key-value parsing for UCX configuration
```
Comment (yosefe) [inline NEWS:39]: should we mention we allow to specify configurations per platform type? see #9742

File: NEWS [hunk 5]
```diff
@@ -11,9 +11,98 @@
...
+* Added a script to use for squashing commits after PR approval
+* Added support for DPU cross-gvmi daemon in UCX perftest
+#### Java
+* Added support for EP local socket address API in JUCX
+#### Build
+* Added Address Sanitizer support
+* Added a helper shell script to run static checks
+#### AZP
+* Added Address Sanitizer job to CI
+* Disabled re-run of all tests when Valgrind tests fail
```
Comment (yosefe) [inline NEWS:55]: Replaced Valgrind tests by Address Sanitizer

File: NEWS [hunk 6]
```diff
@@ -11,9 +11,98 @@
...
+* Fixed worker interface deactivation while still in use by endpoints
+#### RDMA CORE (IB, ROCE, etc.)
+* Disabled device memory if atomics are not available
+* Fixed indirect keys creation for MT registered memory
+* Fixed KSM start address value when creating export key 
+* Fixed DCI pool index to support maximum of 16 pools
+* Fixed atomic rkey issue when using imported memory
+* Fixed crash due to unsupported SRQ capability
+#### GPU (CUDA, ROCM)
+* Removed unused environment variable RCACHE_ADDR_ALIGN from ROCM transport
```
Comment (yosefe) [inline NEWS:86]: ROCm

File: NEWS [hunk 7]
```diff
@@ -11,9 +11,98 @@
...
+* Fixed usage of cuda device 0 when no context is active
+* Removed error handling support from CUDA IPC transport
+* Fixed allocation of unaligned CUDA memory
+#### Shared Memory
+* Fixed occasional crash when shm_unlink fails during interface initialization
+#### UCS
+* Fixed system device distance calculation for devices with different root
```
Comment (yosefe) [inline NEWS:93]: Fixed system device distance calculation for devices on different PCIe root

File: NEWS [hunk 8]
```diff
@@ -11,9 +11,98 @@
...
+#### Tests
+* Fixed test failures when GPU is present but disabled
+* Fixed Active Message hanging issue in ucp_client_server
+* Fixed potential crash due to redundant munmap call in ucp mmap tests 
+* Increased timeout of Valgrind tests
+#### Java
+* Fixed failures in Java tests by waiting for send requests completion
+* Fixed JVM segfault in Java tests when gdrcopy driver is not loaded
+#### Packaging
+* Disabled Go bindings in Debian package
```
Comment (yosefe) [inline NEWS:105]: AFAIR there were also some fixes in Go build

File: NEWS [hunk 9]
```diff
@@ -11,9 +11,98 @@
...
+* Added RMA_ZCOPY_MAX_SEG_SIZE configuration parameter to allow modifying segment size for RMA-ZCOPY protocols
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for protocol variants
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols 
+#### RDMA CORE (IB, ROCE, etc.)
+* Added UCX_REVERSE_SL env var to allow configuring reverse SL and set default to UCX_IB_SL value
```
Comment (yosefe) [inline NEWS:29]: Introduced the UCX_REVERSE_SL environment variable to configure reverse SL for DC transport. By default, it uses UCX_IB_SL.

File: NEWS [hunk 10]
```diff
@@ -11,9 +11,98 @@
...
+* Enabled short protocol for non-host memory types on empty messages
+* Improved performance estimation for empty messages by removing non-relevant overheads
+* Added RMA_ZCOPY_MAX_SEG_SIZE configuration parameter to allow modifying segment size for RMA-ZCOPY protocols
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for protocol variants
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols
```
Comment (yosefe) [inline NEWS:27]: Add to UCP:
* Enable ODP on Grace platforms by default

File: NEWS (hunk 4 above)
Comment (shasson5) [reply NEWS:39]: Line 59:
`* Added support for filtering configuration sections by platform type`

File: NEWS (hunk 5 above)
Comment (yosefe) [reply NEWS:55]: it was done in steps, but currently valgrind is completely disabled: https://github.com/openucx/ucx/blob/eae754ee3be8865a6ba37008efe84ce8f572623b/buildlib/pr/tests.yml#L10

Comment (shasson5) [reply NEWS:55]: and if so also removes line 100:
`* Increased timeout of Valgrind tests`
as it is not relevant anymore

Comment (yosefe) [reply NEWS:55]: > and if so also removes line 100: `* Increased timeout of Valgrind tests` as it is not relevant anymore

Right, should remove it as well

Comment (shasson5) [reply NEWS:55]: also should remove line 55:
`* Added Address Sanitizer job to CI`

File: NEWS (hunk 9 above)
Comment (shasson5) [reply NEWS:29]: maybe remove the second sentence? (as it is too big)

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@brminich can you pls review as well?
@shasson5 pls see the merge conflict..

File: NEWS [hunk 11]
```diff
@@ -11,9 +11,98 @@
...
+## 1.17.0-rc2 (May 16, 2024)
+### Features:
+#### UCP
+* Improved performance calculation for rendezvous protocol
+* Enabled short protocol for non-host memory types on empty messages
+* Improved performance estimation for empty messages by removing non-relevant overheads
+* Added RMA_ZCOPY_MAX_SEG_SIZE configuration parameter to allow modifying segment size for RMA-ZCOPY protocols
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for protocol variants
```
Comment (brminich) [inline NEWS:23]: do we really need to mention this? This is some refactoring not visible to the users

File: NEWS [hunk 12]
```diff
@@ -11,9 +11,98 @@
...
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols
+* Enable ODP on Grace platforms by default
```
Comment (brminich) [inline NEWS:28]:
```suggestion
* Enabled ODP on Grace platforms by default 
```

File: NEWS [hunk 13]
```diff
@@ -11,9 +11,98 @@
...
+* Added support for protocol variants
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols
+* Enable ODP on Grace platforms by default 
+#### RDMA CORE (IB, ROCE, etc.)
+* Introduced the UCX_REVERSE_SL environment variable to configure reverse SL for DC transport. By default, it uses UCX_IB_SL.
+* Added support for GID auto-detection in Floating LID based routing
+* Imporved performance of RC transport when hardware tag matching is not available
```
Comment (brminich) [inline NEWS:32]: Improved
also what is the change mentioned here?

File: NEWS [hunk 14]
```diff
@@ -11,9 +11,98 @@
...
+#### GPU (CUDA, ROCM)
+* Added support for oneAPI Level-Zero library for intel GPUs
+#### UCS
+* Added support for rcache dynamic region alignment
+* Added dynamic bitmap data structure
+* Added support for advanced key-value parsing for UCX configuration
+* Added piecewise linear function data structure
+* Added support for allocating dynamic arrays on stack
+#### Tests
+* Added wire compatibility tests for v1.16
```
Comment (brminich) [inline NEWS:44]: do we have to mention it?

File: NEWS [hunk 15]
```diff
@@ -11,9 +11,98 @@
...
+* Added support for DPU cross-gvmi daemon in UCX perftest
+#### Java
+* Added support for EP local socket address API in JUCX
+#### Build
+* Added Address Sanitizer support
+* Added a helper shell script to run static checks
+#### AZP
+* Replaced Valgrind tests by Address Sanitizer 
+* Added Ubuntu 22.04 docker image testing
+* Updated gdrcopy module version
```
Comment (brminich) [inline NEWS:57]: afair we have not mentioned such changes before

File: NEWS [hunk 16]
```diff
@@ -11,9 +11,98 @@
...
+#### Configuration
+* Added support for filtering configuration sections by platform type
+* Added configuration file with section for Grace Hopper
```
Comment (brminich) [inline NEWS:60]: pls add emplty line below

Comment (shasson5) [reply NEWS:60]: seems in 1.16 there's no space line (line 169)

Comment (brminich) [reply NEWS:60]: ok

File: NEWS (hunk 13 above)
Comment (shasson5) [reply NEWS:32]: https://github.com/openucx/ucx/pull/9644

Comment (brminich) [reply NEWS]: i'd remove that - it is internal refactoring needed for implementing gga support. No perf improvements expected with this patch

**Review by brminich — APPROVED** [review]

**Review by yosefe — DISMISSED** [review]

File: NEWS [hunk 17]
```diff
@@ -11,9 +11,93 @@
 ### Features:
 ### Bugfixes:
 
-## 1.17.0-rc1 (May 16, 2024)
-TBD
-
+## 1.17.0-rc2 (May 16, 2024)
+### Features:
+#### UCP
+* Improved performance calculation for rendezvous protocol
```
Comment (shamisp) [inline NEWS:17]: Improved the accuracy of rendezvous protocol performance estimation ?

File: NEWS [hunk 18]
```diff
@@ -11,9 +11,93 @@
...
+* Enabled short protocol for non-host memory types on empty messages
+* Improved performance estimation for empty messages by removing non-relevant overheads
```
Comment (shamisp) [inline NEWS:19]: Improved the accuracy ?

File: NEWS [hunk 19]
```diff
@@ -11,9 +11,93 @@
...
+* Added RMA_ZCOPY_MAX_SEG_SIZE configuration parameter to allow modifying segment size for RMA-ZCOPY protocols
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols
+* Enabled ODP on Grace platforms by default
```
Comment (shamisp) [inline NEWS:27]: On demand paging for Nvidia's Grace

File: NEWS [hunk 20]
```diff
@@ -11,9 +11,93 @@
...
+* Added RMA_ZCOPY_MAX_SEG_SIZE configuration parameter to allow modifying segment size for RMA-ZCOPY protocols
+* Added support for separate intra/inter-node rendezvous thresholds
+* Added support for minimal fragment size in rendezvous protocol
+* Added support for resetting request during send operation
+* Added UCX_PROTO_OVERHEAD configuration variable to allow setting protocol overheads
+* Improved performance for combined Active Message/RMA scenarios by separating them to different lanes
+* Added support for device staging buffers in pipeline protocols
+* Enabled ODP on Grace platforms by default 
+#### RDMA CORE (IB, ROCE, etc.)
+* Introduced the UCX_REVERSE_SL environment variable to configure reverse SL for DC transport. By default, it uses UCX_IB_SL.
```
Comment (shamisp) [inline NEWS:29]: What is reverse SL ?

File: NEWS [hunk 21]
```diff
@@ -11,9 +11,93 @@
...
+* Added support for GID auto-detection in Floating LID based routing
+* Added support for multithreading KSM registration of unaligned buffers
+* Added IB_SEND_OVERHEAD and MM_[SEND|RECV]_OVERHEAD configuration variables
+#### GPU (CUDA, ROCM)
+* Added support for oneAPI Level-Zero library for intel GPUs
```
Comment (shamisp) [inline NEWS:34]: intel -> Intel

File: NEWS [hunk 22]
```diff
@@ -11,9 +11,93 @@
...
+* Added piecewise linear function data structure
+* Added support for allocating dynamic arrays on stack 
+#### Tools
+* Added support for device memory allocation in UCX perftest
+* Added a script to use for squashing commits after PR approval
+* Added support for DPU cross-gvmi daemon in UCX perftest
+#### Java
+* Added support for EP local socket address API in JUCX
+#### Build
+* Added Address Sanitizer support
```
Comment (shamisp) [inline NEWS:48]: Address Sanitizer -> address sanitizer

File: NEWS [hunk 23]
```diff
@@ -11,9 +11,93 @@
...
+* Added a helper shell script to run static checks
+#### AZP
+* Replaced Valgrind tests by Address Sanitizer
```
Comment (shamisp) [inline NEWS:51]: Replaced Valgrind tests with address sanitizer tool ?

**Review by shamisp — CHANGES_REQUESTED** [review]

**Comment by shamisp** [comment]
Please do not forget to update `Authors` file with the names of new contributors.

File: NEWS (hunk 19 above)
Comment (gleon99) [reply NEWS:27]: on-demand

File: NEWS (hunk 20 above)
Comment (gleon99) [reply NEWS:29]: @shamisp 
That's a HW feature, documented in other places. Are you sure more details are needed here, in addition to what is already mentioned?

**Comment by gleon99** [comment]
> Please do not forget to update `Authors` file with the names of new contributors.
Already done & merged in a separate PR.

**Comment by shasson5** [comment]
@shamisp can review?

File: NEWS (hunk 20 above)
Comment (shamisp) [reply NEWS:29]: We need to use here a description that external users can understand. I don't believe anybody will know what is UCX_REVERSE_SL

Comment (yosefe) [reply NEWS:29]: How about:
"Introduced the UCX_REVERSE_SL environment variable to configure priority level for DC transport inbound traffic."

Comment (shamisp) [reply NEWS:29]: I like @yosefe  proposal

Comment (shamisp) [reply NEWS:29]: Actually, current version is fine as well. I like that it calls out default behaviour.

**Review by shamisp — APPROVED** [review]

**Comment by yosefe** [comment]
@shasson5 can you pls squash?

**Review by yosefe — APPROVED** [review]

**Pull request merged**